src/Bicep.LangServer/Handlers/BicepBuildCommandHandler.cs

```diff
@@ -0,0 +1,125 @@
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Bicep.Core;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Emit;
+using Bicep.Core.Semantics;
+using Bicep.Core.Text;
+using Bicep.Core.Workspaces;
+using Bicep.LanguageServer.CompilationManager;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    // This handler is used to generate compiled .json file for given a bicep file path.
+    // It returns build succeeded/failed message, which can be displayed approriately in IDE output window
+    public class BicepBuildCommandHandler : ExecuteTypedResponseCommandHandlerBase<string, string>
+    {
+        private readonly ICompilationManager CompilationManager;
+
+        public BicepBuildCommandHandler(ICompilationManager compilationManager, ISerializer serializer)
+            : base(LanguageConstants.Build, serializer)
+        {
+            CompilationManager = compilationManager;
+        }
+
+        public override Task<string> Handle(string bicepFilePath, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(bicepFilePath))
+            {
+                throw new ArgumentException("Invalid input file");
+            }
+
+            DocumentUri documentUri = DocumentUri.FromFileSystemPath(bicepFilePath);
+            string buildOutput = GenerateCompiledFileAndReturnBuildOutputMessage(bicepFilePath, documentUri);
+
+            return Task.FromResult(buildOutput);
+        }
+
+        private string GenerateCompiledFileAndReturnBuildOutputMessage(string bicepFilePath, DocumentUri documentUri)
+        {
+            StringBuilder sb = new StringBuilder();
+
+            string? compiledFilePath = GetCompiledFilePath(bicepFilePath, sb, out string bicepFile, out string compiledFile);
+
+            if (string.IsNullOrWhiteSpace(compiledFilePath))
+            {
+                throw new ArgumentException($"Invalid input file");
+            }
+
+            CompilationContext? context = CompilationManager.GetCompilation(documentUri);
+
+            if (context is null)
+            {
+                throw new InvalidOperationException($"Unable to get compilation context");
+            }
+
+            SemanticModel semanticModel = context.Compilation.GetEntrypointSemanticModel();
+            KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile = context.Compilation.GetAllDiagnosticsByBicepFile()
+                .FirstOrDefault(x => x.Key.FileUri == documentUri.ToUri());
+
+            if (diagnosticsByFile.Value.Any())
+            {
+                AppendDiagnosticsMessage(diagnosticsByFile, bicepFile, sb);
+                return sb.ToString();
+            }
+
+            using (FileStream fileStream = new FileStream(compiledFilePath, FileMode.Create, FileAccess.ReadWrite, FileShare.ReadWrite))
```

Comment:
>FileStream [](http://example.com/codeflow?start=47&length=10)

A couple of thoughts here:
1. If the user has a JSON file with the same name at the location that wasn't generated by Bicep before, we should fail the Build command. Otherwise, we could wipe out the user's data. (A workspace may not be under version control). This is the first write operation we're going to do in the user's workspace that can't be undone with Ctrl+Z like codefixes can. 

I think we have to be careful with this. The best option is probably to check if the template contains Bicep _generator metadata. It's not perfect, but should cover the most cases. (The library that can deserialize templates is already available in Bicep.)

2. `FileShare.ReadWrite` means that some other process is allowed to read the file while we're writing, which probably won't do anything useful. We should probably just use the default `FileShare` setting for now.

Reply:
could we have just a popup saying if user wants to override existing file? Usually we'd use build to preview the arm template and this command would be used by more advanced users that wouldn't mind hitting enter one more time.

More user-friendly command would be ability to deploy template to Azure directly but that's another issue.

Reply:
The typical workflow for the feature will be make a change, click build, make a change, click build. It could get pretty annoying to have to click yes every time if someone does that after every change. However, if they only click Build once in a while, a confirmation prompt makes sense. It is also the safest option because it's user's responsibility to protect their own files and make correct choices. It's also much simpler to implement.

Reply:
Added check to see if file already exists and fail the build as part of this commit:
https://github.com/Azure/bicep/commit/daa24e89c632c399b92f0de41c98fcea77b930eb

Let me know if I am missing something

Reply:
@majastrz , as discussed offline, added logic to replace file only if template contains bicep generator metadata:
https://github.com/Azure/bicep/commit/e7e05ab2faec4b5977345b5f5d5e672d4d59d28e

Let me know if I am missing something.